### PR TITLE
Watcher updates

### PR DESCRIPTION
**Pull request by eikenb**
3 tweaks not tied to a feature but wanted to get cleaned up.  Each of the 3 things has it's own commit to make them easier to digest.

Remove unused watcher.Remove and supporting code.
Clear cache entry during sweep step.
refine parameter type requirements, Notifier->IDer

### CODE REVIEW

File: watcher.go [hunk 1]
```diff
@@ -420,9 +419,15 @@ func (tp trackedPair) markCacheAccessed() trackedPair {
 	return tp
 }
 
+// IDer an interface that supports and ID
+type IDer interface {
+	ID() string
+}
```
Comment (lornasong) [inline watcher.go:425]: Interesting pattern! Noting this down to use some day in the future!

**Review by lornasong — APPROVED** [review]
nice ✨ yay for deleting code!

File: watcher.go (hunk 1 above)
Comment (findkim) [reply watcher.go:425]: The interface "-er" suffix has always been amusing to me

Comment (eikenb) [reply watcher.go:425]: I've been trying to be stricter with my inputs by defining more specific Interface types as needed. I feel there is probably a better point of balance that is more specific/detailed than I've done in the past.

And while I've just accepted the "-er" naming... it comes at a cost (eg. Watcherer, Renderer... :grimacing:)

**Pull request merged**